pkg/util/util.go

```diff
@@ -114,9 +114,55 @@ func GetLocalEndpoint(clusterID, backend string, backendConfig map[string]string
 		endpoint.Spec.PublicIP = publicIP
 	}
 
+	endpoint.Spec.HealthCheckIP, err = getCNIInterfaceIPAddress(clusterCIDR)
+
+	if err != nil {
+		return types.SubmarinerEndpoint{}, fmt.Errorf("Error getting CNI Interface IP address: %v", err)
+	}
+
 	return endpoint, nil
 }
 
+func getCNIInterfaceIPAddress(clusterCIDRs []string) (string, error) {
+	for _, clusterCIDR := range clusterCIDRs {
+		_, clusterNetwork, err := net.ParseCIDR(clusterCIDR)
```

Comment:
Isn't this code shared with route agent somewhere? I remember we already had that this somewhere.

@sridhargaddam do you remember?

Reply:
Yes there is a similar code but not exactly the same. 
https://github.com/submariner-io/submariner/blob/master/pkg/routeagent/controllers/route/driver.go#L15

Reply:
Ok, we will need to refactor this later, this code is also undergoing some changes, so better not to rely on it now. 

Can you add a TODO: to handle de-duplication of code/finding common parts with the route agent ?

Reply:
Added TODO